app/helpers/sprangular/application_helper.rb

```diff
@@ -86,25 +78,10 @@ def cached_templates
     end
 
     def cached_templates_for_dir(files, dir)
-      root = Sprangular::Engine.root
+      logical_paths = templates_environment.each_logical_path("#{dir}/**")
 
-      files = Dir[root + "app/assets/templates/#{dir}/**"].inject(files) do |hash, path|
-        asset_path = asset_path path.gsub(root.to_s + "/app/assets/templates/", "")
-        local_path = "app/assets/templates/" + asset_path
-
-        hash[asset_path.gsub(/.slim$/, '')] = Tilt.new(path).render.html_safe if !File.exists?(local_path)
-
-        hash
-      end
-
-      Dir["app/assets/templates/#{dir}/**"].inject(files) do |hash, path|
-        sprockets_path = path.gsub("app/assets/templates/", "")
-
-        asset_path = asset_path(sprockets_path).
-          gsub(/^\/app\/assets\/templates/, '/assets').
-          gsub(/.slim$/, '')
-
-        hash[asset_path] = Rails.application.assets.find_asset(sprockets_path).body.html_safe
+      logical_paths.inject(files) do |hash, path|
+        hash[asset_path(path)] = assets_environment[path].body.html_safe
```

Comment:
Note here that `assets_environment` is used to process the template, while `templates_environment` was used build the set of templates. `templates_environment` will not register any non-standard template engines (e.g. Slim, Haml, etc.), and is used only to isolate template lookups to Sprangular and the host app. Leave the template processing to the asset-pipeline!